MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -856,6 +859,10 @@ namespace WPEFramework {
                         /* we set this to false */
                         g_is_critical_maintenance="false";
 
+                        if(m_thread.joinable()){
```

Comment:
Updating based on discussion with @michaelamal. - This won't cause a deadlock because, only one Maintenance (only one thread) can run at a time. If a second "startMaintenance" API  is called by Application then it will go to else and exit from there.